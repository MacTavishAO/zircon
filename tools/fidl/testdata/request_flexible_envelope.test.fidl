library fidl.test.json;

flexible union FlexibleFoo {
    1: string s;
    2: int32 i;
};

union StrictFoo {
    1: string s;
    2: int32 i;
};

protocol Protocol {
    RequestStrictResponseFlexible(StrictFoo s) -> (FlexibleFoo f);
    RequestFlexibleResponseStrict(FlexibleFoo s) -> (StrictFoo f);
};
