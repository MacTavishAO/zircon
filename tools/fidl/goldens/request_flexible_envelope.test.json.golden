{
  "version": "0.0.1",
  "name": "fidl.test.json",
  "library_dependencies": [],
  "bits_declarations": [],
  "const_declarations": [],
  "enum_declarations": [],
  "experimental_resource_declarations": [],
  "interface_declarations": [
    {
      "name": "fidl.test.json/Protocol",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 13,
        "column": 10,
        "length": 8
      },
      "methods": [
        {
          "ordinal": 8264567585134801538,
          "name": "RequestStrictResponseFlexible",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 14,
            "column": 5,
            "length": 29
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/StrictFoo",
                "nullable": false
              },
              "name": "s",
              "location": {
                "filename": "request_flexible_envelope.test.fidl",
                "line": 14,
                "column": 45,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix0",
          "maybe_request_type_shape_v1": {
            "inline_size": 40,
            "alignment": 8,
            "depth": 2,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/FlexibleFoo",
                "nullable": false
              },
              "name": "f",
              "location": {
                "filename": "request_flexible_envelope.test.fidl",
                "line": 14,
                "column": 64,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix1",
          "maybe_response_type_shape_v1": {
            "inline_size": 40,
            "alignment": 8,
            "depth": 2,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": true,
            "is_resource": false
          },
          "is_composed": false
        },
        {
          "ordinal": 4801763909694511442,
          "name": "RequestFlexibleResponseStrict",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 15,
            "column": 5,
            "length": 29
          },
          "has_request": true,
          "maybe_request": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/FlexibleFoo",
                "nullable": false
              },
              "name": "s",
              "location": {
                "filename": "request_flexible_envelope.test.fidl",
                "line": 15,
                "column": 47,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_request_payload": "fidl.test.json/SomeLongAnonymousPrefix2",
          "maybe_request_type_shape_v1": {
            "inline_size": 40,
            "alignment": 8,
            "depth": 2,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": true,
            "is_resource": false
          },
          "has_response": true,
          "maybe_response": [
            {
              "type": {
                "kind": "identifier",
                "identifier": "fidl.test.json/StrictFoo",
                "nullable": false
              },
              "name": "f",
              "location": {
                "filename": "request_flexible_envelope.test.fidl",
                "line": 15,
                "column": 64,
                "length": 1
              },
              "field_shape_v1": {
                "offset": 16,
                "padding": 0
              }
            }
          ],
          "maybe_response_payload": "fidl.test.json/SomeLongAnonymousPrefix3",
          "maybe_response_type_shape_v1": {
            "inline_size": 40,
            "alignment": 8,
            "depth": 2,
            "max_handles": 0,
            "max_out_of_line": 4294967295,
            "has_padding": true,
            "has_flexible_envelope": false,
            "is_resource": false
          },
          "is_composed": false
        }
      ]
    }
  ],
  "service_declarations": [],
  "struct_declarations": [
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix0",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 14,
        "column": 34,
        "length": 13
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/StrictFoo",
            "nullable": false
          },
          "name": "s",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 14,
            "column": 45,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix1",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 14,
        "column": 51,
        "length": 15
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/FlexibleFoo",
            "nullable": false
          },
          "name": "f",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 14,
            "column": 64,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix2",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 15,
        "column": 34,
        "length": 15
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/FlexibleFoo",
            "nullable": false
          },
          "name": "s",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 15,
            "column": 47,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/SomeLongAnonymousPrefix3",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 15,
        "column": 53,
        "length": 13
      },
      "anonymous": true,
      "members": [
        {
          "type": {
            "kind": "identifier",
            "identifier": "fidl.test.json/StrictFoo",
            "nullable": false
          },
          "name": "f",
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 15,
            "column": 64,
            "length": 1
          },
          "field_shape_v1": {
            "offset": 0,
            "padding": 0
          }
        }
      ],
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "table_declarations": [],
  "union_declarations": [
    {
      "name": "fidl.test.json/FlexibleFoo",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 3,
        "column": 16,
        "length": 11
      },
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "name": "s",
          "type": {
            "kind": "string",
            "nullable": false
          },
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 4,
            "column": 15,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": false,
          "name": "i",
          "type": {
            "kind": "primitive",
            "subtype": "int32"
          },
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 5,
            "column": 14,
            "length": 1
          }
        }
      ],
      "strict": false,
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": true,
        "is_resource": false
      }
    },
    {
      "name": "fidl.test.json/StrictFoo",
      "location": {
        "filename": "request_flexible_envelope.test.fidl",
        "line": 8,
        "column": 7,
        "length": 9
      },
      "members": [
        {
          "ordinal": 1,
          "reserved": false,
          "name": "s",
          "type": {
            "kind": "string",
            "nullable": false
          },
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 9,
            "column": 15,
            "length": 1
          }
        },
        {
          "ordinal": 2,
          "reserved": false,
          "name": "i",
          "type": {
            "kind": "primitive",
            "subtype": "int32"
          },
          "location": {
            "filename": "request_flexible_envelope.test.fidl",
            "line": 10,
            "column": 14,
            "length": 1
          }
        }
      ],
      "strict": true,
      "type_shape_v1": {
        "inline_size": 24,
        "alignment": 8,
        "depth": 2,
        "max_handles": 0,
        "max_out_of_line": 4294967295,
        "has_padding": true,
        "has_flexible_envelope": false,
        "is_resource": false
      }
    }
  ],
  "type_alias_declarations": [],
  "declaration_order": [
    "fidl.test.json/StrictFoo",
    "fidl.test.json/FlexibleFoo",
    "fidl.test.json/Protocol"
  ],
  "declarations": {
    "fidl.test.json/Protocol": "interface",
    "fidl.test.json/SomeLongAnonymousPrefix0": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix1": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix2": "struct",
    "fidl.test.json/SomeLongAnonymousPrefix3": "struct",
    "fidl.test.json/FlexibleFoo": "union",
    "fidl.test.json/StrictFoo": "union"
  }
}
